### ISSUE TITLE
Deleted/active recipients flag

### ISSUE THREAD

**Issue by adria0** [Giveth/vaultcontroller#35]
Consider to add a flag `active` in `struct Recipient`(like `struct Spender`) instead using `activationTime` == `0` as deletion flag

**Comment by jbaylina** [comment]
It but it would take more gas...

**Comment by GriffGreen** [comment]
gas optimization is something I want to do some time on this contract... i don't know if its best done now or after the first run of tests.... but low gas is a huge priority... mostly cause i think I will end up using these contracts a lot and am very cheap ;-)

**Comment by dipenjoshi** [comment]
i'm mostly worried about `out of gas` exceptions. being cheap on gas is 2nd priority.

**Comment by GriffGreen** [comment]
I agree about the out of gas annoyances! This is something that mostly needs to be fixed in `estimateGas()` on the UI side, but it has been a big problem for us

**Comment by jbaylina** [comment]
Here, the discusion is about "gas optimization" <-> "code readability" .

**Comment by adria0** [comment]
Proposal:

- Finish tests (& fix code bugs)
- Make special tests only to check gas usage for specific scenarios (which ones?)
- Make proposals for "gas optimization" <-> "code readability" equilibrium. Check changes against gas tests and then decide.

**Comment by GriffGreen** [comment]
sounds good.
